File: python-amberum-taf/tools/webdriver_factory.py
import os

from selenium import webdriver


class WebdriverFactory:

    def get_webdriver(self):
        return self.get_chromedriver()
        # return self.get_firefoxdriver()
        # return self.get_edgedriver()

    def get_chromedriver(self):
        driver_path = os.environ.get("WEBDRIVER_CHROME_PATH")
        return webdriver.Chrome(executable_path=driver_path)

    def get_firefoxdriver(self):
        driver_path = os.environ["WEBDRIVER_FIREFOX_PATH"]
        return webdriver.Firefox(executable_path=driver_path)

    def get_edgedriver(self):
        driver_path = os.environ["WEBDRIVER_EDGE_PATH"]
        return webdriver.Edge(executable_path=driver_path)

File: java-amberum-taf/src/test/java/com/amberum/tools/WebDriverFactory.java
package com.amberum.tools;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.firefox.FirefoxOptions;

import java.util.concurrent.TimeUnit;

public class WebDriverFactory {

    public static WebDriver getWebDriver() {
        //WebDriver driver = getFirefoxDriver();
        WebDriver driver = getChromeDriver();
        // EdgeDriver driver = getEdgeDriver();

        driver.manage().timeouts().implicitlyWait(10, TimeUnit.SECONDS);
        return driver;
    }

    public static FirefoxDriver getFirefoxDriver() {
        System.setProperty("webdriver.gecko.driver", System.getenv("WEBDRIVER_CHROME_PATH"));
        FirefoxOptions firefoxOptions = new FirefoxOptions();
        FirefoxDriver driver = new FirefoxDriver(firefoxOptions);
        return driver;
    }

    public static ChromeDriver getChromeDriver() {
        System.setProperty("webdriver.chrome.driver", System.getenv("WEBDRIVER_CHROME_PATH"));
        ChromeDriver driver = new ChromeDriver();
        return driver;
    }

    public static EdgeDriver getEdgeDriver() {
        System.setProperty("webdriver.edge.driver", System.getenv("WEBDRIVER_EDGE_PATH"));
        EdgeDriver driver = new EdgeDriver();
        return driver;
    }

}
